ee/server/service/calendar.go

```diff
@@ -186,7 +186,7 @@ func (svc *Service) processCalendarEvent(ctx context.Context, eventDetails *flee
 	if updated && event != nil {
 		// Event was updated, so we need to save it
 		_, err = svc.ds.CreateOrUpdateCalendarEvent(ctx, event.UUID, event.Email, event.StartTime, event.EndTime, event.Data,
-			event.TimeZone, eventDetails.HostID, fleet.CalendarWebhookStatusNone)
+			*event.TimeZone, eventDetails.HostID, fleet.CalendarWebhookStatusNone)
```

Comment:
The `CreateOrUpdateCalendarEvent` method should take the pointer (it may panic here otherwise).

Reply:
Won't `event.TimeZone` here never be `nil`, since the event returned from `GetAndUpdateEvent` now updates any `nil` timezone for the event before returning it?

Reply:
I get that that's a brittle state of affairs to rely on so makes sense to change regardless, just double-checking my understanding

Reply:
This is so that we don't have to make assumptions around what `GetAndUpdateEvent` returns. (If in the future it returns a `nil` `event.TimeZone` then we prefer a NULL in the DB instead of a panic.)